repo/blob/s3/s3_storage.go

```diff
@@ -181,6 +181,7 @@ func (s *s3Storage) putBlob(ctx context.Context, b blob.ID, data blob.Bytes, opt
 		StorageClass:    storageClass,
 		RetainUntilDate: retainUntilDate,
 		Mode:            retentionMode,
+		PartSize:		 s.Options.UploadPartSize,
```

Comment:
Does it make sense to have this configurable per `putBlob()` as well? If y'all think that'd be useful, we can follow up with a PR in the future.

Reply:
I wouldn't do it per blob, since it's hard to plumb through all the places we'd need this. Provider-level config is much easier to reason about.